Video room

yanivel opened the pull request:
- Refactor grid item to remove deduplication.
- Clean up some of the styling and remove unused functions/variables.

This is in preparation to make changing layouts easier.

igala commented on galaxy_mobile/lib/utils/switch_page_helper.dart:
```diff
@@ -83,22 +83,17 @@ class SwitchPageHelper {
-            if (feed != null && feed.isNotEmpty) feed["videoSlot"] = index;
-          })
-        : null;
+    oldVideoFeeds.forEach((feed) {
```
@yanivel you removed the check for empty, was it on purpose?

yanivel replied:
Yes, we don't need to check if it's not empty since forEach will not do anything, so I simplified the code.

igala commented:
Not sure ,
This check was done in purpose by me due to empty case exception.

> On 10 Mar 2022, at 5:10, yanivel ***@***.***> wrote:
> 
> ﻿
> @yanivel commented on this pull request.
> 
> In galaxy_mobile/lib/utils/switch_page_helper.dart:
> 
> > -    //       if (feed != null) {feed["videoSlot"] = -1}
> -    //     });
> -    oldVideoFeeds.isNotEmpty
> -        ? oldVideoFeeds.forEach((feed) {
> -            if (feed != null) {
> -              feed["videoSlot"] = null;
> -            }
> -          })
> -        : null;
> -    newVideoFeeds.isNotEmpty
> -        ? newVideoFeeds.forEach((feed) {
> -            var index = newVideoFeeds.indexOf(feed);
> -            if (feed != null && feed.isNotEmpty) feed["videoSlot"] = index;
> -          })
> -        : null;
> +    oldVideoFeeds.forEach((feed) {
> Yes, we don't need to check if it's not empty since forEach will not do anything, so I simplified the code.
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> Triage notifications on the go with GitHub Mobile for iOS or Android. 
> You are receiving this because you were assigned.

yanivel commented:
I don't think such an exception can happen for running forEach on an empty list (it could have been something in the function inside the forEach), but I re-added the check for isNotEmpty anyway. I also added it below on some other forEach that did not have this check in head.
I changed it to a simple "if" instead of a ternary operation.

igala commented:
Ok, thx.


> On 10 Mar 2022, at 5:55, yanivel ***@***.***> wrote:
> 
> ﻿
> I don't think such an exception can happen for running forEach on an empty list, but I re-added the check for isNotEmpty. I also added it below on some other forEach that did not have this check in head.
> I changed it to a simple "if" instead of a ternary operation.
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> Triage notifications on the go with GitHub Mobile for iOS or Android. 
> You are receiving this because you were assigned.
